### PR TITLE
Force core if stop gently failed

### PR DESCRIPTION
**Pull request by bhalevy**
If gently stopping the node failed retry with signal.SIGQUIT

Aborting is intended to generate a core dump
so the reason the node didn't stop normally can be studied.

Refs scylladb/scylla#4384 scylladb/scylla#5020

Test: `bootstrap_test:TestBootstrap.start_stop_test`
and the same with instrumenting scylla to hang on shutdown

### CODE REVIEW

File: ccmlib/scylla_node.py [hunk 1]
```diff
@@ -514,26 +529,31 @@ def stop(self, wait=True, wait_other_notice=False, other_nodes=None, gently=True
                     except OSError:
                         pass
 
+            if not wait and not wait_other_notice:
+                return True
+
+            time.sleep(1)
```
Comment (fruch) [inline ccmlib/scylla_node.py]: Why sleeping here ?

File: ccmlib/scylla_node.py [hunk 2]
```diff
@@ -514,26 +529,31 @@ def stop(self, wait=True, wait_other_notice=False, other_nodes=None, gently=True
...
+                        except OSError:
+                            pass
+                    self.wait_until_stopped(wait_seconds)
+                if self.is_running():
+                    raise NodeError("Problem stopping node %s" % self.name)
+
             if wait_other_notice:
                 for node, mark in marks:
                     node.watch_log_for_death(self, from_mark=mark)
-            else:
```
Comment (fruch) [inline ccmlib/scylla_node.py]: Why switch the order ? first killing the node then waiting for others ?

Comment (bhalevy) [reply ccmlib/scylla_node.py]: Others won't notice if the node hangs during shutdown,
so we want to force kill it if needed and then look if the others noticed.

**Comment by bhalevy** [comment]
Successful test [here](https://jenkins.scylladb.com/view/master/job/scylla-master/job/manual-and-scheduled-tests/job/byo_build_tests_dtest/632/console)

**Pull request merged**